proto/buf/alpha/registry/v1alpha1/recommendation.proto

```diff
@@ -18,12 +18,42 @@ package buf.alpha.registry.v1alpha1;
 
 import "google/protobuf/timestamp.proto";
 
+// The resource types of the recommendation.
+enum ResourceType {
+  RESOURCE_TYPE_UNSPECIFIED = 0;
+  RESOURCE_TYPE_REPOSITORY = 1;
+  RESOURCE_TYPE_TEMPLATE = 2;
+}
+
+message Resource {
+  oneof resource {
+    // The resource is a `repository`.
+    string repository_id = 1;
+    // The resource is a `template`.
+    string template_id = 2;
+  }
+}
+
+message Recommendation {
+  // unique, immutable
+  string id = 1;
+  // unique, immutable
+  Resource resource = 2;
```

Comment:
What do you think about inlining this message instead of making a separate message for it?
```suggestion
  oneof resource {
    // The resource is a `repository`.
    string repository_id = 2;
    // The resource is a `template`.
    string template_id = 3;
  }
```